### PR TITLE
Expand ebay-icon capabilities to stamp custom inline icons

### PR DESCRIPTION
**Pull request by dsathyakumar**
## Description
Expand `ebay-icon` capabilities to stamp custom inline icons.

## Context
- Currently `<ebay-icon/>` stamps inline SVG icons that are available in `/symbols` folder within it
- More often, there are repeated custom icons given, which currently cannot utilize the one time stamping inside `<body>`, as done by `<ebay-icon/>`.
- Now, adding a custom icon is as simple as
>`<ebay-icon type="inline" name="my-awesome-icon" custom="true"/>` 

(this causes the transformer to look within the project root for a `/symbols/my-awesome-icon.marko`)

## Screenshots
image

**PS**: I'm a newbie to this project 🙂, pls excuse if the PR doesn't meet your standards!

### CODE REVIEW

File: src/components/ebay-icon/index.js [hunk 1]
```diff
@@ -26,6 +26,8 @@ function init() {
 }
 
 function getTemplateData(state, input, out) {
+    const custom = input.custom || 'false';
+    const isCustom = custom === 'true';
```
Comment (senthilp) [inline src/components/ebay-icon/index.js]: Looks like `custom` is a boolean attribute. As per our [guidelines](https://medium.com/@senthil_hi/building-a-ui-component-in-2017-and-beyond-1f6d5c4d464#f7e3), if an attribute is boolean, just the presence of the attribute on an element represents the true value. Can you please update it accordingly? This should also be reflected in docs.

File: src/components/ebay-icon/template.marko [hunk 2]
```diff
@@ -6,6 +6,11 @@
     <svg w-bind ${data.htmlAttributes} ${data.accessibilityAttributes} focusable="false" class=data.classes>
         <defs w-id="defs" w-body if(data.renderDefs)/>
         <title if(data.accessibilityText) id=data.titleId>${data.accessibilityText}</title>
-        <use xlink:href="#icon-${data.name}"/>
+        <if(data.isCustom)>
+            <use iscustom=data.isCustom xlink:href="#${data.name}"/>
+        </if>
+        <else>
+            <use iscustom=data.isCustom xlink:href="#icon-${data.name}"/>
```
Comment (senthilp) [inline src/components/ebay-icon/template.marko]: Do you still need `iscustom=data.isCustom` in the else block?

**Comment by yomed** [comment]
The stamping functionality is definitely useful outside of `<ebay-icon>`, but I think that while we're in the official `<ebay-icon>` component, we should limit ourselves to the design-sanctioned list of icons.

If we wanted to share the stamping, I think we would need to abstract out the stamping to its own utility/component.

**Comment by dsathyakumar** [comment]
🤔 seems strange that we would want to limit the list of icons & have functionality in two places.

**Comment by seangates** [comment]
@dsathyakumar It does not seem strange when we want to limit the icons to a very specific set.

That said, and to re-iterate what @yomed said, we should abstract the stamping capability. Then `<ebay-icon>` could stamp a _limited set of icons_, and any other component could stamp their own set (or not limit the icon set, whichever they prefer).

**Comment by seangates** [comment]
Created issue #285 and closing.

**Pull request closed without merging**